### PR TITLE
Remove upper limit for ChainLevelHelper.GetNextHeader

### PR DESCRIPTION
**Pull request by asdacap**
## Changes:
- Remove ChainLevelHelper.GetNextHeader.

## Types of changes

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [X] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [X] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by MarekM25** [comment]
our unit tests are failing

**Comment by asdacap** [comment]
Fixed

**Review by MarekM25 — APPROVED** [review]

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs [hunk 1]
```diff
@@ -110,6 +110,13 @@ public class MergeBlockDownloader : BlockDownloader
 
             bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
             {
+                if (!_beaconPivot.BeaconPivotExists())
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs:113]: This condition is wrong. We can have situation when we don't have beacon pivot and we want to sync.

Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs:113]: what do you think about Math.Min(ProcessDestination, BestKnownNumber) >= BestKnownBeaconNumber?

Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs:113]: Imagine that you have timeout in NewPayloadHandler. For the next payload we insert it without cache and return SYNCING. We don't have beaconPivot in this situation.

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**